crates/solvers/src/infra/dex/mod.rs

```diff
@@ -20,18 +22,19 @@ impl Dex {
         &self,
         order: &dex::Order,
         slippage: &dex::Slippage,
-        gas_price: auction::GasPrice,
+        auction: &Auction,
```

Comment:
Not passing in an `Auction` was also an intentional thing I had from before. The rationale here is you end up passing in data that may be "inconsistent", and it is unclear from the caller's perspective what the expectations are:
- What happens if there is no order in the auction that is equivalent to the order parameter?
- What do you do with all the liquidity?
- Does the callee check the deadline?
- etc.

Essentially, it makes the function operate on a lot more data that it actually cares about and creates some ambiguity on what the function actually does with the additional data.

Reply:
https://github.com/cowprotocol/services/pull/1513/commits/aeb6e3bc819f272f013c007647b143c54de351a4